src/v3/tutorial/middleware/README.md

```diff
@@ -0,0 +1,87 @@
+---
+description: This tutorial will show you how to use a middleware.
+full_title: Middleware Tutorial
+---
+
+# Middleware Tutorial
+
+This tutorial will show you how to use a middleware. A middleware is used when an application wants to provide a way for other applications to change its behavior.
+
+## Usage
+
+```javascript
+// To add a callback to a middleware group
+core.make('osjs/middleware').add('middleware-group-name', middlewareCallback);
+// OR
+import osjs from 'osjs';
+
+osjs.middleware('middleware-group-name', middlewareCallback);
+
+// To get the list of callbacks in a middleware group
+const middlewareCallbacks = core.make('osjs/middleware').get('middleware-group-name');
+```
```

Comment:
Split up into three examples:

---

Register a callback to middleware defined by a group name:

```javascript
const middlewareFunction = () => {}`
core.make('osjs/middleware')
  .add('middleware-group-name', middlewareCallback);
```

Alternatively you can the global registration function:

```javascript
import osjs from 'osjs';

const middlewareFunction = () => {}`
osjs.middleware('middleware-group-name', middlewareCallback);
```

The get the list of registered middleware from a group:

```javascript
const middlewareCallbacks = core.make('osjs/middleware')
  .get('middleware-group-name'); // => [middlewareFunction]
```